framework/users.py

```diff
@@ -0,0 +1,189 @@
+        set explicity, this will return `None`.
+        """
+        return self.__user_id
+
+
+    def auth_domain(self):
+        """Obtains the user's authentication domain.
+
+        Returns:
+        A string containing the authentication domain. This method is internal and
+        should not be used by client applications.
+        """
+        return self.__auth_domain
+
+
+    def federated_identity(self):
+        """Decommissioned, don't use.
+
+        Returns:
+        A string containing the federated identity of the user. If the user is not
+        a federated user, `None` is returned.
+        """
+        return self.__federated_identity
+
+
+    def federated_provider(self):
+        """Decommissioned, don't use.
+
+        Returns:
+        A string containing the federated provider. If the user is not a federated
+        user, `None` is returned.
+        """
+        return self.__federated_provider
+
+
+    def __unicode__(self):
+        return six_subset.text_type(self.nickname())
+
+    def __str__(self):
+        return str(self.nickname())
+
+    def __repr__(self):
+        values = []
+        if self.__email:
+            values.append("email='%s'" % self.__email)
+        if self.__federated_identity:
+            values.append("federated_identity='%s'" % self.__federated_identity)
+        if self.__user_id:
+            values.append("_user_id='%s'" % self.__user_id)
+        return 'users.User(%s)' % ','.join(values)
+
+    def __hash__(self):
+        if self.__federated_identity:
+            return hash((self.__federated_identity, self.__auth_domain))
+        else:
+            return hash((self.__email, self.__auth_domain))
+
+    def __cmp__(self, other):
+        if not isinstance(other, User):
+            return NotImplemented
+        if self.__federated_identity:
+            return cmp((self.__federated_identity, self.__auth_domain),
+                        (other.__federated_identity, other.__auth_domain))
+        else:
+            return cmp((self.__email, self.__auth_domain),
+                        (other.__email, other.__auth_domain))
+
+    def get_current_user():
+        """Retrieves information associated with the user that is making a request.
+
+        Returns:
+
+        """
+        try:
+            return User()
+        except UserNotFoundError:
+            return None
+
+
+    def is_current_user_admin():
+        """Specifies whether the user making a request is an application admin.
+
+        Because administrator status is not persisted in the datastore,
+        `is_current_user_admin()` is a separate function rather than a member function
+        of the `User` class. The status only exists for the user making the current
+        request.
+
+        Returns:
+        `True` if the user is an administrator; all other user types return `False`.
+        """
+        return (os.environ.get('USER_IS_ADMIN', '0')) == '1'
```

Comment:
I imagine that this environment variable is set by GAE based on a GAE session cookie.  With Google Sign-In, it will probably never be present.  You can leave this as-is or even just return the constant False.  I figure out another way to represent system administrators and land that change separately.

Reply:
Okay, I will add a comment stating the same and leave it as-is.

Reply:
Done